### DISCUSSION TITLE
GraphQL HTTP POST-Request Error; "Problems parsing JSON"

### DISCUSSION THREAD

**Discussion by Gisgar3** [community/community#24480] [Apps, API and Webhooks]
<p> Hello, I am having some trouble parsing <strong>JSON </strong> data in the  <strong>GitHub GraphQL API</strong>. I am unsure if I am not writing the <strong>JSON</strong> in the correct format, or there’s another problem stemming from it. Below are photos of my command-prompt and my Visual Studio Code window: </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9189d43877852323d70d4768f8c30213a83da18c.png" data-download-href="/uploads/short-url/kLuwT4LtF7lvXjpNPS8wWDxEcsA.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24739483/181084842-db29e6ca-7d23-4edc-a5d4-f9001250c246.png" alt="image.png" data-base62-sha1="kLuwT4LtF7lvXjpNPS8wWDxEcsA" width="690" height="164" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9189d43877852323d70d4768f8c30213a83da18c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">1199×285 18.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/9b36877d06bb4740f4129faaf53ed94edef71b77.png" data-download-href="/uploads/short-url/m94PQKdlhKPFlo6o4H3EkHR4LVt.png?dl=1" title="image.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24739483/181084843-9fe13ab1-3f2b-4170-8fc1-96dfb5a8f002.png" alt="image.png" data-base62-sha1="m94PQKdlhKPFlo6o4H3EkHR4LVt" width="583" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/9b36877d06bb4740f4129faaf53ed94edef71b77_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image.png</span><span class="informations">674×578 3.16 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p> I appreciate any help that I can get for this error; have been very busy recently and am now able to write a post about this ongoing issue. Thanks!</p>

**Comment by lee-dohm** [comment]
<p>It looks like you forgot the word <code>query</code> from the first line of your GraphQL query structure, before the opening curly brace. This is understandable because there’s a lot of use of the word <code>query</code> in your code example :grinning: So the contents should probably be:</p>
<pre><code>query { repository(owner: ...
</code></pre>
<p>I hope that helps!</p>

**Comment by Gisgar3** [comment]
<p>I have tried adding <em>query</em> like your example shows but that method still results in the same parsing error. Any other ideas? Thanks!</p>

**Comment by lee-dohm** [comment]
<p>Have you tried putting the query <a href="https://developer.github.com/v4/explorer/" rel="nofollow noopener">into the Explorer tool in our developer site</a> to see what error message it shows?</p>

**Comment by Gisgar3** [comment]
<p>Yep, tried the Explorer before writing the code just to make sure the query worked properly. Not sure what could be causing this error to occur.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Ok, so then it isn’t the query text itself but the method of calling the endpoint that is the problem. The only guess I have there is that you seem to be using the <a href="https://github.com/request/request" rel="nofollow noopener">JavaScript request library</a> and the documentation seems to indicate that the <code>form</code> parameter of the <code>post</code> method adds the data you supply as querystring parameters, rather than adding the text to the body. Try supplying the query contents to the <code>body</code> parameter of the <code>post</code> method instead.</p>

**Comment by Gisgar3** [comment]
<p>Thank you, this  <strong>did fix</strong> my problem. Alongside that, I also made sure to set the <code>json</code> parameter to <em>true</em> within the statement!</p>
